ivy_tests/test_ivy/test_functional/test_core/test_manipulation.py

```diff
@@ -288,39 +274,48 @@ def test_reshape(
 
 # roll
 @settings(
-    deadline=750,
+    deadline=1250,
+    suppress_health_check=(HealthCheck.data_too_large,),  # jax.roll is very slow
 )
 @given(
     dtype_value=helpers.dtype_and_values(
         available_dtypes=ivy_np.valid_dtypes,
         shape=st.shared(helpers.get_shape(min_num_dims=1), key="value_shape"),
     ),
     shift=helpers.dtype_and_values(
-        available_dtypes=[ivy.int32, ivy.int64],
+        available_dtypes=[ivy.int32],
         max_num_dims=1,
         min_dim_size=st.shared(
-            helpers.ints(min_value=1, max_value=2147483647), key="shift_length"
+            helpers.array_values(dtype="int32", shape=(1,))
```

Comment:
could you add a short comment explaining the purpose of the `map` and `filter` calls.

Reply:
Removed `map` and `filter` calls to something more readable. Added comment above the data generation to explain what is happening and why